### PR TITLE
[tune] use isinstance instead of type for TBXLogger

### PR DESCRIPTION
**Pull request by sumanthratna**
## Why are these changes needed?

isinstance instead of type

## Related issue number

n/a

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]

File: python/ray/tune/logger.py [hunk 1]
```diff
@@ -219,17 +219,16 @@ def on_result(self, result):
 
         for attr, value in flat_result.items():
             full_attr = "/".join(path + [attr])
-            if type(value) in VALID_SUMMARY_TYPES and not np.isnan(value):
+            if isinstance(value, VALID_SUMMARY_TYPES) and not np.isnan(value):
```
Comment (richardliaw) [inline python/ray/tune/logger.py:222]: this needs to cast VALID_SUMMARY_TYPES to tuple (tests are failing)

Comment (sumanthratna) [reply python/ray/tune/logger.py:222]:
```suggestion
            if (isinstance(value, tuple(VALID_SUMMARY_TYPES))
                and not np.isnan(value)):
```

File: python/ray/tune/logger.py [hunk 2]
```diff
@@ -260,7 +260,7 @@ def close(self):
                 scrubbed_result = {
                     k: value
                     for k, value in flat_result.items()
-                    if type(value) in VALID_SUMMARY_TYPES
+                    if isinstance(value, VALID_SUMMARY_TYPES)
```
Comment (sumanthratna) [inline python/ray/tune/logger.py:263]:
```suggestion
                    if isinstance(value, tuple(VALID_SUMMARY_TYPES))
```

File: python/ray/tune/logger.py [hunk 3]
```diff
@@ -219,17 +219,17 @@ def on_result(self, result):
 
         for attr, value in flat_result.items():
             full_attr = "/".join(path + [attr])
-            if type(value) in VALID_SUMMARY_TYPES and not np.isnan(value):
+            if (isinstance(value, tuple(VALID_SUMMARY_TYPES))
+                and not np.isnan(value)):
```
Comment (sumanthratna) [inline python/ray/tune/logger.py:223]:
```suggestion
                    and not np.isnan(value)):
```

**Pull request merged**